app/lib/event_store/active_record_event_repository.rb

```diff
@@ -5,7 +5,7 @@ class ActiveRecordEventRepository < RailsEventStoreActiveRecord::EventRepository
     def build_event_entity(record)
       return nil unless record
 
-      data = (record.data || {}).merge(
+      data = record.data.merge(
```

Comment:
can't data be `nil`? In such case original version will not fail.

Reply:
In the DB NULL values are permitted for the `data` column. But as this field is serialized/deserialized as YAML, when the object is instantiated, it appears that `record.data` is set to an empty hash, if the `data` column value is nil, so it seems fine.
I tested with the following: 
```
class TestEvent < BaseEventStoreEvent
  def self.create
    new(
      metadata: {
        provider_id: 2
      }
     
    )
  end
end

event = TestEvent.create_and_publish!

event_from_repo = EventStore::Repository.find_event(event.event_id)
```

The `data` column is NULL for this event, but `build_event_entity` works fine.

I didn't know about this when reverting the change, but I thought - it used to work before, so it should still work.

In any case, we don't have any events in the code base that would have empty `data`, all have at least something.